### PR TITLE
#156577923 Send Email alert when a job is run too often

### PR DESCRIPTION
**Pull request by rogerokello**
**What does this PR do?**
This PR enables an Email alert to be sent to a user when a job is run too often and creates an icon on the dashboard for alerts that are run too often

**Description of Task to be completed?**
When a job is pinged too early, an Email alert is sent to the user and the status icon of the job changes to red.

**How should this be manually tested?**
1. Create a job with a timeout period of three minutes and grace period of one minute.
2. Ping the job URL to activate it
3. Ping that same URL again in less than a minute
4. An email alert should be sent to your mailbox
5. The icon on the dashboard next to the should change to red indicating it is running too often

**Any background context you want to provide?**
1. Previously a user was not able to be sent an E-mail when a job was pinged before the timeout period minus grace period. Now they can.
2. Previously jobs that were run too often would not be shown on the dashboard. Now they can

**What are the relevant pivotal tracker stories?**
[#156577923](https://www.pivotaltracker.com/story/show/156577923)

### CODE REVIEW

**Comment by kasigazibrian** [comment]
@rogerokello Could you add a link to your pivotal tracker story id in the pull request

**Comment by rogerokello** [comment]
@kasigazibrian Thank you so much for your input. I have tried to implement it in the pull request.

File: hc/api/tests/test_ping.py [hunk 1]
```diff
@@ -103,3 +103,15 @@ def test_csrf_client_head(self):
         self.assertEqual(r.status_code, 200)
         self.assertEqual(self.check.status, "up")
 
+    def test_it_handles_check_running_often(self):
+        #ping first time to get it up and running
+        r = self.client.get("/ping/%s/" % self.check.code)
+        assert r.status_code == 200
+
+        # ping second time to get it to too often
+        r = self.client.get("/ping/%s/" % self.check.code)
```
Comment (Rwothoromo) [inline hc/api/tests/test_ping.py]: How are we asserting that the second ping actually occurs?

**Comment by emugaya** [comment]
@rogerokello Rebase your branch with `develop` and make sure that test coverage doesn't decrease

**Comment by rogerokello** [comment]
@emugaya Alright I'll be sure to do that now

**Review by emugaya — APPROVED** [review]
LGTM

**Comment by louicoder** [comment]
@rogerokello great work. however, you have 13 commits. you may have to squash them into one. otherwise , great work

**Pull request closed without merging**